ironic/resource_ironic_deployment.go

```diff
@@ -85,8 +101,45 @@ func resourceDeploymentCreate(d *schema.ResourceData, meta interface{}) error {
 
 	d.SetId(d.Get("node_uuid").(string))
 
+	userData := d.Get("user_data")
+	userDataURL := d.Get("user_data_url")
+	userDataCaCert := d.Get("user_data_url_ca_cert")
+
+	// Send full ignition, if the URL is specified
+	if userDataURL != "" {
+		caCertPool := x509.NewCertPool()
+		transport := &http.Transport{}
+
+		if userDataCaCert != "" {
+			caCert, err := base64.StdEncoding.DecodeString(userDataCaCert.(string))
+			if err != nil {
+				return fmt.Errorf("could not decode user_data_url_ca_cert: %s", err)
+			}
+			caCertPool.AppendCertsFromPEM(caCert)
+
+			transport.TLSClientConfig = &tls.Config{RootCAs: caCertPool}
+		} else {
+			// Disable certificate verification
+			transport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+		}
+
+		client := retryablehttp.NewClient()
```

Comment:
Neat!

Reply:
Do we think the defaults are enough? Looks like 4 tries, max 30 seconds.

I think maybe for the openshift case, the ignition will usually be available by the time the ironic API is, so it's maybe not a big deal.


https://github.com/hashicorp/go-retryablehttp/blob/eaff869a704e190d92996f23c23a4037cf2b0770/client.go#L47-L49

Reply:
Yeah, I didn't mess with the defaults and wanted to wait until the review to add them. Not sure what are a good defaults for our purposes are? How about 10 retries and maxwait of 60s?

Reply:
The default might be fine for our purposes, I'm not really sure.